ta/pkcs11/src/pkcs11_attributes.c

```diff
@@ -1041,6 +1051,8 @@ enum pkcs11_rc check_created_attrs_against_processing(uint32_t proc_id,
 	 */
 	switch (proc_id) {
 	case PKCS11_PROCESSING_IMPORT:
+	case PKCS11_CKM_AES_ECB_ENCRYPT_DATA:
+	case PKCS11_CKM_AES_CBC_ENCRYPT_DATA:
```

Comment:
I think we should address derivation by ciphering weakness before this change is merged, to be shure it's addressable without ugly hacks.
What about this:
```c
enum pkcs11_rc check_created_attrs_against_processing(uint32_t proc_id,
							      struct obj_attrs *head)
{
	/*
	 * Processings that do not create secrets are not expected to call
	 * this function which would panic.
	 */
	switch (proc_id) {
	case PKCS11_CKM_AES_ECB_ENCRYPT_DATA:
	case PKCS11_CKM_AES_CBC_ENCRYPT_DATA:
		if (get_bool(CKA_ENCRYPT, head)) {
			/*
			 * Intentionally refuse to proceed despite PKCS#11
			 * specification v2.40 and v3.0 do not expect this
			 * behavior.
			 */
			return PKCS11_CKR_FUNCTIONAL_FAILED;
		}

		fallthrough;
	case PKCS11_PROCESSING_IMPORT:
		assert(check_attr_bval(proc_id, head, PKCS11_CKA_LOCAL, false));
		break;
	(...)

Reply:
I think the idea is good but my sugggestion is not at the right place.
should be in `check_parent_attrs_against_processing()`, not in `check_created_attrs_against_processing()`.

Reply:
I agree it would be good to add the restriction. Thanks for the suggestion.

 I will do the change, update the testcases templates also and update the PR.